content/bn/docs/concepts/architecture/garbage-collection.md

```diff
@@ -0,0 +1,136 @@
+
+
+### ফরগ্রাউন্ড ক্যাসকেডিং ডিলিশন {#foreground-deletion}
+
+ফরগ্রাউন্ড ক্যাসকেডিং ডিলিশনে, আপনি যে মালিকানা অবজেক্টটি মুছে ফেলছেন সেটি প্রথমে একটি *ডিলিশন ইন প্রগ্রেস* অবস্থায় চলে যায়। এই অবস্থায়, মালিকানা অবজেক্টের সাথে নিম্নলিখিত ঘটনা ঘটে:
+
+* Kubernetes API সার্ভার অবজেক্টের `metadata.deletionTimestamp` ক্ষেত্রটিকে সেই সময়ে সেট করে যখন অবজেক্টটি মুছে ফেলার জন্য চিহ্নিত করা হয়।
+* Kubernetes API সার্ভার এছাড়াও `metadata.finalizers` ক্ষেত্রটিকে `foregroundDeletion` এ সেট করে।
+* অবজেক্টটি Kubernetes API-এর মাধ্যমে দৃশ্যমান থাকে যতক্ষণ না মুছে ফেলার প্রক্রিয়া সম্পূর্ণ হয়।
+
+মালিকানা অবজেক্টটি ডিলিশন ইন প্রগ্রেস অবস্থায় চলে গেলে, কন্ট্রোলার নির্ভরশীল অবজেক্টগুলি মুছে ফেলে। সমস্ত নির্ভরশীল অবজেক্ট মুছে ফেলার পর কন্ট্রোলার মালিকানা অবজেক্টটি মুছে ফেলবে। এই মুহূর্তে, অবজেক্টটি Kubernetes API-এ আর দৃশ্যমান থাকে না।
+
+ফরগ্রাউন্ড ক্যাসকেডিং ডিলিশন চলাকালীন, একমাত্র নির্ভরশীলরা যারা মালিক মুছে ফেলার প্রতিবন্ধকতা সৃষ্টি করে তা হল যাদের `ownerReference.blockOwnerDeletion=true` ক্ষেত্র রয়েছে। আরো জানার জন্য [ফরগ্রাউন্ড ক্যাসকেডিং ডিলিশন ব্যবহার করুন।](/bn/docs/tasks/administer-cluster/use-cascading-deletion/#use-foreground-cascading-deletion)
+
+### ব্যাকগ্রাউন্ড ক্যাসকেডিং ডিলিশন {#background-deletion}
+
+ব্যাকগ্রাউন্ড ক্যাসকেডিং ডিলিশনে, Kubernetes API সার্ভার অবজেক্টটি অবিলম্বে মুছে ফেলে এবং কন্ট্রোলার ব্যাকগ্রাউন্ডে নির্ভরশীল অবজেক্টগুলি পরিষ্কার করে। স্বাভাবিকভাবে, Kubernetes ব্যাকগ্রাউন্ড ক্যাসকেডিং ডিলিশন ব্যবহার করে যতক্ষণ না আপনি ম্যানুয়ালি ফরগ্রাউন্ড ডিলিশন ব্যবহার করেন বা নির্ভরশীল অবজেক্টগুলি অরফান না করেন।
+
+আরো জানার জন্য [ব্যাকগ্রাউন্ড ক্যাসকেডিং ডিলিশন ব্যবহার করুন।](/bn/docs/tasks/administer-cluster/use-cascading-deletion/#use-background-cascading-deletion)
+to learn more.
+
+### অরফানড ডিপেনডেন্টস  {#orphaned-dependents}
+
+যখন কুবার্নেটিস একটি মালিক অবজেক্ট মুছে ফেলে, তখন বাকি থাকা ডিপেনডেন্টসগুলোকে অরফান অবজেক্ট বলা হয়। ডিফল্টভাবে, কুবার্নেটিস ডিপেনডেন্ট অবজেক্টগুলো মুছে ফেলে। এই আচরণটি কিভাবে ওভাররাইড করবেন তা জানতে, দেখুন [মালিক অবজেক্ট মুছে ফেলুন এবং অরফান ডিপেনডেন্টস](/bn/docs/tasks/administer-cluster/use-cascading-deletion/#set-orphan-deletion-policy)
+
+## অব্যবহৃত কনটেইনার এবং চিত্রগুলির গার্বেজ কালেকশন {#containers-images}
+
+{{<glossary_tooltip text="kubelet" term_id="kubelet">}} প্রতি দুই মিনিটে অব্যবহৃত চিত্রগুলির গার্বেজ কালেকশন এবং প্রতি মিনিটে অব্যবহৃত কনটেইনারগুলির গার্বেজ কালেকশন করে। বাহ্যিক গার্বেজ কালেকশন টুলস ব্যবহার থেকে বিরত থাকুন, কারণ এটি কিউবিলেটের আচরণ ভেঙে ফেলতে পারে এবং এমন কনটেইনারগুলি মুছে ফেলতে পারে যা থাকার উচিত।
+অব্যবহৃত কনটেইনার এবং চিত্রগুলির গার্বেজ কালেকশন জন্য বিকল্প কনফিগার করতে, কিউবিলেটটি একটি [কনফিগারেশন ফাইল](/bn/docs/tasks/administer-cluster/kubelet-config-file/) ব্যবহার করে কনফিগার করুন এবং গার্বেজ কালেকশন সম্পর্কিত প্যারামিটারগুলি পরিবর্তন করুন [`KubeletConfiguration`](/bn/docs/reference/config-api/kubelet-config.v1beta1/)
```

Comment:
Use the English version as reference `[কনফিগারেশন ফাইল](/docs/tasks/administer-cluster/kubelet-config-file/)` and ```[`KubeletConfiguration`](/docs/reference/config-api/kubelet-config.v1beta1/)```